src/pymodaq_plugins_keithley/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Keithley27XX.py

```diff
@@ -0,0 +1,248 @@
+            self.controller.set_mode(value)
+        elif self.panel == 'REAR':
+            self.controller.configuration_sequence()
+            value = 'SCAN_' + self.settings.child('Keithley_Params', 'rearpanel', 'rearmode').value()
+            self.channels_in_selected_mode = self.controller.set_mode(value)
+            print('Channels to plot :', self.channels_in_selected_mode)
+        print('DAQ_viewer command sent to keithley visa driver :', value)
+
+        self.status.initialized = True
+        self.status.controller = self.controller
+
+        return self.status
+
+    def close(self):
+        """Terminate the communication protocol"""
+        self.controller.close()
+        print('communication ended successfully')
+
+    def grab_data(self, Naverage=1, **kwargs):
+        """Start a grab from the detector
+
+        :param Naverage: Number of hardware averaging (if hardware averaging is possible, self.hardware_averaging should be set to
+            True in class preamble and you should code this implementation)
+        :type Naverage: int
+
+        :param kwargs: others optionals arguments
+        :type kwargs: dict
+        """
+        # ACQUISITION OF DATA
+        if self.panel == 'FRONT':
+            data_tot = self.controller.data()
+            data_measurement = data_tot[1]
+        elif self.panel == 'REAR':
+            print('current mode', self.controller.current_mode)
+            channels_in_selected_mode = self.channels_in_selected_mode[1:-1].replace('@', '')
+            Chan_to_plot = []
+            data_tot = self.controller.data()
+            data_measurement = data_tot[1]
+            for i in range(len(channels_in_selected_mode.split(','))):
+                Chan_to_plot.append('Channel ' + str(channels_in_selected_mode.split(',')[i]))
+            # Affect each value to the corresponding channel
+            dict_chan_value = dict(zip(channels_in_selected_mode.split(','), data_measurement))
+
+            print('Chan :', Chan_to_plot)
+            print('dict chan value : ', dict_chan_value)
+        print('Data tot :', data_tot)
+        print('Data measurement :', data_measurement)
+
+        # Dictionary linking channel's modes to physical quantities
+        dict_label_mode = {'VOLT:DC': 'Voltage', 'VOLT:AC': 'Voltage', 'CURR:DC': 'Current', 'CURR:AC': 'Current',
+                           'RES': 'Resistance', 'FRES': 'Resistance', 'FREQ': 'Frequency', 'TEMP': 'Temperature'}
+
+        # EMISSION OF DATA
+        # When reading the scan_list, data are displayed and exported grouped by mode
+        if self.controller.reading_scan_list == False:
+            Label = dict_label_mode[self.controller.current_mode]
+            if self.panel == 'FRONT':
+                labels = 'Front input'
+            elif self.panel == 'REAR':
+                labels = [Chan_to_plot[i] for i in range(len(Chan_to_plot))]
+            data = DataToExport(name='keithley',
```

Comment:
try to use meaning full date for variable such as dte or data_dte

Reply:
Done using dte